lib/syskit/cli/log_runtime_archive_main.rb

```diff
@@ -114,6 +114,49 @@ def transfer_server( # rubocop:disable Metrics/ParameterLists
                               implicit_ftps)
             end
 
+            desc "watch_ensure_free_space", "watches the ensure free space process"
+            option :period,
+                   type: :numeric, default: 10, desc: "polling period in seconds"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            option :free_space_low_limit,
+                   type: :numeric, default: 5_000, desc: "start deleting files if \
+                    available space is below this threshold (threshold in MB)"
+            option :free_space_freed_limit,
+                   type: :numeric, default: 25_000, desc: "stop deleting files if \
+                    available space is above this threshold (threshold in MB)"
+            def watch_ensure_free_space(source_dir)
+                loop do
+                    ensure_free_space(source_dir)
+
+                    puts "Ensured free space in #{source_dir}, " \
+                         "sleeping #{options[:period]}s"
+                    sleep options[:period]
+                end
+            end
+
+            desc "ensure_free_space", "ensures there is free space, if not, start \
+                                       deleting files"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            option :free_space_low_limit,
+                   type: :numeric, default: 5_000, desc: "start deleting files if \
+                    available space is below this threshold (threshold in MB)"
+            option :free_space_freed_limit,
+                   type: :numeric, default: 25_000, desc: "stop deleting files if \
+                    available space is above this threshold (threshold in MB)"
+            def ensure_free_space(source_dir)
```

Comment:
For the log transfer, you need to recurse one level inside folders. The `source_dir` should be the root where log directories are created, while in the log archiving we're looking at the archive dir.

Actually, there is also the issue of how we decide which file to delete. In the archive case, we assumed names would be alphabetically sorted in time (they're not, read on). In this case, we deal with arbitrary files.

ensure_free_space should now be using the modification time. Which would also solve the bug on the archive case, where `xxxx.10.tar` would be deleted before `xxxxx.2.tar`

Reply:
Just so we are clear, I also used this modification time when deciding what order I would ensure free space in the source dir, meaning that older directories would be affected first by deletion.